src/main/kotlin/no/item/xp/plugin/GenerateTypeScriptTask.kt

```diff
@@ -1,27 +1,96 @@
 package no.item.xp.plugin
 
+import arrow.core.Either
 import java.io.File
+import java.nio.file.Path
+import java.nio.file.Paths
 import javax.inject.Inject
+import javax.xml.parsers.DocumentBuilderFactory
+import no.item.xp.plugin.models.GeneratedField
+import no.item.xp.plugin.models.XmlFile
+import no.item.xp.plugin.models.XmlType
+import no.item.xp.plugin.parser.parse
+import no.item.xp.plugin.util.GenerateInterfaceName
 import org.gradle.api.DefaultTask
 import org.gradle.api.tasks.TaskAction
+import org.w3c.dom.Document
+import java.lang.Exception
+import javax.xml.parsers.DocumentBuilder
 
+@Suppress("UNUSED_VARIABLE")
 open class GenerateTypeScriptTask @Inject constructor(private val extension: GenerateTypeScriptExtension) : DefaultTask() {
 
+  private val docBuilder: DocumentBuilder = DocumentBuilderFactory.newInstance().newDocumentBuilder()
+
     @TaskAction
     fun generateTypeScript() {
-      val files = findXmlFiles()
-      println("Found XML files: ")
-      println("--------------------")
-      files.forEach { println(it) }
-      // 2.send dem til en parser
-      // 3.send resultatet til en writer
-      // 4.exit
-    }
-
+      val files: Sequence<String> = findXmlFiles()
+      files.forEach { parseFile(it) }
+  }
   private fun findXmlFiles(): Sequence<String> {
     return File(extension.fileDir)
       .walk()
-      .filter { it.isFile && it.extension == "xml" }
+      .filter { it.isFile }
       .map { it.absolutePath }
   }
+
+  private fun parseFile(filePath: String) {
+    val xmlType: Either<Throwable, XmlType> = getXmlType(filePath)
+    var xmlTypeValue: XmlType? = null
+    when(xmlType){
+      is Either.Right -> {
+        xmlTypeValue = xmlType.b
+      }
+    }
+    if(xmlTypeValue!=null) {
+      val xmlFile: XmlFile = generateXmlFile(filePath, xmlTypeValue)
+      val document: Document = readXml(xmlFile)
+      parse(document)
+
+      val interfaceName: String? = GenerateInterfaceName(File(filePath))
+      val xml: Either<Throwable, Sequence<GeneratedField>> = parse(document)
+      /*if (interfaceName != null) {
+      TypeScriptWriter(interfaceName, xmlType, xml)
+    }*/
+    }
+  }
```

Comment:
Add a return type to every function. Here we should just do a `map` for `flatMap` to only handle the right value of the `Either`. We never need to unpack it with a `when`.

```suggestion
  private fun parseFile(filePath: String): Either<Throwable, Sequence<GeneratedField>> {
    return getXmlType(filePath)
      .flatMap {
        val xmlFile = generateXmlFile(filePath, it)
        val document = readXml(xmlFile)
        parse(document)
      }
  }
```